app/utils/combat.ts

```diff
@@ -4,11 +4,14 @@ export function damageTarget(state: GameState, target: AttackTarget, damage: num
         return
     }
     target.health = Math.max(0, target.health - damage);
-    if (target.health <= 0) {
-        // remove the object from the state when it dies.
+    if (target.health <= 0 && target.objectType !== 'nexus') {
+        // remove the object from the state, if not a 'nexus' when it dies.
```

Comment:
For your next iteration, let's add a flag to the top level of state like `nexusIsDestroyed?: boolean` and then instead of skipping all of this for the nexus, let's set that flag to true here. Then in other code you can read that flag to decide whether to use the normal render/update logic or the new render/update logic for when the nexus is destroyed.